auth0/src/main/java/com/auth0/android/provider/WebAuthProvider.java

```diff
@@ -57,17 +60,107 @@
+        private CustomTabsOptions ctOptions;
+
+        LogoutBuilder(Auth0 account) {
+            this.account = account;
+
+            //Default values
+            this.scheme = "https";
+            this.values = new HashMap<>();
+        }
+
+        /**
+         * When using a Custom Tabs compatible Browser, apply this customization options.
+         *
+         * @param options the Custom Tabs customization options
+         * @return the current builder instance
+         */
+        public LogoutBuilder withCustomTabsOptions(@NonNull CustomTabsOptions options) {
+            this.ctOptions = options;
+            return this;
+        }
+
+        /**
+         * Specify a custom Scheme to use on the Return To Uri. Default scheme is 'https'.
+         *
+         * @param scheme to use in the Callback Uri.
+         * @return the current builder instance
+         */
+        public LogoutBuilder withScheme(@NonNull String scheme) {
+            String lowerCase = scheme.toLowerCase();
+            if (!scheme.equals(lowerCase)) {
+                Log.w(TAG, "Please provide the scheme in lowercase and make sure it's the same configured in the intent filter. Android expects the scheme in lowercase");
+            }
+            this.scheme = scheme;
+            return this;
+        }
+
+        /**
+         * Whether to also clear the session from the identity provider when possible or not.
+         *
+         * @param federatedLogout true to clear the session on the identity provider's site.
+         * @return the current builder instance
+         */
+        public LogoutBuilder useFederatedLogout(@NonNull boolean federatedLogout) {
+            if (federatedLogout) {
+                this.values.put(KEY_FEDERATED, "1");
+            } else {
+                this.values.remove(KEY_FEDERATED);
+            }
+            return this;
+        }
+
+        /**
+         * Request the user session to be cleared. When successful, the callback will get invoked
+         *
+         * @param context  to run the log out
+         * @param callback to invoke when log out is successful
+         */
+        public void start(Context context, BaseCallback<Void, Auth0Exception> callback) {
+            managerInstance = null;
+            if (account.getLogoutUrl() == null) {
+                Throwable cause = new IllegalArgumentException("Auth0 logout URL not properly set. This can be related to an invalid domain.");
+                final Auth0Exception ex = new Auth0Exception("Cannot perform web log out", cause);
+                callback.onFailure(ex);
+                return;
+            }
+
+            if (!hasBrowserAppInstalled(context.getPackageManager())) {
+                Throwable cause = new ActivityNotFoundException("No Browser application installed.");
+                final Auth0Exception ex = new Auth0Exception("Cannot perform web log out", cause);
+                callback.onFailure(ex);
```

Comment:
I wanted to differentiate this immediate failures by providing 2 different classes on the Cause.

-When the domain value was wrong and the logout url couldn't be generated, throw an `Auth0Exception` with `IllegalArgumentException` cause.
-When there is no browser app to handle this, throw an `Auth0Exception` with `ActivityNotFoundException` cause.


This is here to avoid creating a new type (class) of exception. Users could differentiate the cause by doing `instanceof` on it.